src/utils/calculations.ts

```diff
@@ -404,6 +406,66 @@ export const calcPoolStatePreview = (previewInputs: PoolStatePreviewInputs): Poo
     ? new BigNumber(1)
     : expectedLongBalance.div(expectedShortBalance);
 
+    return ({
+        expectedLongBalance,
+        expectedShortBalance,
+        totalNetPendingLong,
+        totalNetPendingShort,
+        expectedLongSupply,
+        expectedShortSupply,
+        expectedLongTokenPrice,
+        expectedShortTokenPrice,
+        expectedOraclePrice: movingOraclePriceAfter,
+        expectedSkew
+    })
+
+}
+
+/**
+ * calculates the expected state of the pool after applying the given pending commits to the given pool state
+ * @param previewInputs
+ * @returns the expected state of the pool
+ */
+export const calcPoolStatePreview = (previewInputs: PoolStatePreviewInputs): PoolStatePreview => {
+    const {
+        leverage,
+        fee,
+        longBalance,
+        shortBalance,
+        longTokenSupply,
+        shortTokenSupply,
+        pendingLongTokenBurn,
+        pendingShortTokenBurn,
+        lastOraclePrice,
+        currentOraclePrice,
+        pendingCommits,
+    } = previewInputs;
+    // calc balancers after fees
+    const { longBalance: longBalanceAfterFee, shortBalance: shortBalanceAfterFee } = calcBalancesAfterFees(longBalance, shortBalance, fee);
+
+    // calc balances after value transfer
+    const valueTransfer = calcNextValueTransfer(lastOraclePrice, currentOraclePrice, leverage, longBalanceAfterFee, shortBalanceAfterFee)
+    const longBalanceAfterPriceChange = longBalanceAfterFee.plus(valueTransfer.longValueTransfer)
+    const shortBalanceAfterPriceChange = shortBalanceAfterFee.plus(valueTransfer.shortValueTransfer)
```

Comment:
i think this wil result in 1 too many value transfer simulations since it will also do one per iteration when applying commits

Reply:
ease, doesnt break tests then